api/src/main/java/org/pmiops/workbench/dataset/DataSetServiceImpl.java

```diff
@@ -815,6 +815,58 @@ private String getQualifiedColumnName(Domain currentDomain, String columnName) {
             + "head results.P2.assoc");
   }
 
+  @Override
+  public List<String> generateHailDemoCode(String qualifier) {
+    final String phenotypeFilename = "phenotypes_annotations_" + qualifier + ".txt";
+    final String cohortQualifier = "cohort_" + qualifier;
+    final String cohortVcfFilename = cohortQualifier + ".vcf";
+    final String cohortMtFilename = cohortQualifier + ".mt";
+
+    return ImmutableList.of(
+        "import subprocess, os\n"
+            + "import random\n"
+            + "\n"
+            + "# Creating phenotype annotations file\n"
+            + "phenotypes_table = []\n"
+            + "for person_id in person_ids:\n"
+            + "    person_id = person_id\n"
+            + "    phenotype_1 = random.randint(0, 2) # Change this value to what makes sense for your research by looking through the dataset(s)\n"
+            + "    phenotype_2 = random.randint(0, 2) # Change this value as well or remove if you are only processing one phenotype \n"
+            + "    phenotypes_table.append([person_id, phenotype_1, phenotype_2])\n"
+            + "\n"
+            + "cohort_phenotypes = pandas.DataFrame(phenotypes_table,columns=[\"sample_name\", \"phenotype1\", \"phenotype2\"]) \n"
+            + "cohort_phenotypes.to_csv('"
+            + phenotypeFilename
+            + "', index=False, sep='\\t')\n"
+            + "\n"
+            + "subprocess.run([\"gsutil\", \"cp\", \""
+            + phenotypeFilename
+            + "\", os.environ['WORKSPACE_BUCKET']])",
+        "import hail as hl\n"
+            + "import os\n"
+            + "from hail.plot import show\n"
+            + "\n"
+            + "hl.plot.output_notebook()\n"
+            + "bucket = os.environ['WORKSPACE_BUCKET']\n"
+            + "hl.import_vcf(f'{bucket}/"
+            + cohortVcfFilename
+            + "').write(f'{bucket}/"
+            + cohortMtFilename
+            + "')\n"
+            + "table = hl.import_table(f'{bucket}/"
+            + phenotypeFilename
+            + "', types={'sample_name': hl.tstr}, impute=True, key='sample_name')\n"
+            + "\n"
+            + "mt = hl.read_matrix_table(f'{bucket}/"
+            + cohortMtFilename
+            + "');\n"
+            + "mt = mt.annotate_cols(pheno = table[mt.s])\n"
+            + "\n"
+            + "gwas = hl.linear_regression_rows(y=mt.pheno.phenotype1, x=mt.GT.n_alt_alleles(), covariates=[1.0])\n"
```

Comment:
If covariates is something the user should expect to be modifying, then please comment or pull out into a variable